### ISSUE TITLE
hillshade fails on chunked DataArray with cuda but no cupy

### ISSUE THREAD

**Issue by TomAugspurger** [makepath/xarray-spatial#584]
**Describe the bug**

A TypeError is raised when applying hillshade when

1. The DataArray is chunked
2. cuda is available, but cupy isn't installed

```pytb
In [1]: import xarray as xr, numpy as np, xrspatial.hillshade

In [2]: xrspatial.hillshade(xr.DataArray(np.random.randn(10, 10)).chunk())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-8f1525117e3d> in <module>
----> 1 xrspatial.hillshade(xr.DataArray(np.random.randn(10, 10)).chunk())

~/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/xrspatial/hillshade.py in hillshade(agg, azimuth, angle_altitude, name)
    180
    181     # cupy/numba case
--> 182     elif has_cuda() and isinstance(agg.data, cupy.ndarray):
    183         out = _run_cupy(agg.data, azimuth, angle_altitude)
    184

TypeError: isinstance() arg 2 must be a type or tuple of types
````

**Additional context**

https://github.com/makepath/xarray-spatial/blob/cb768bd8375212486069440b2567fe789bb5dc7c/xrspatial/hillshade.py#L6-L11 defines `cupy.ndarray` to be False.

https://github.com/makepath/xarray-spatial/blob/cb768bd8375212486069440b2567fe789bb5dc7c/xrspatial/hillshade.py#L181-L183 passes that `False` to `isinstance`, which is expecting a class.

**Comment by ianthomas23** [comment]
@TomAugspurger Thanks for reporting this. In general we have inconsistent checking and dealing with presence and/or absence of cupy and cuda. Issue #571 is related.

**Comment by thuydotm** [comment]
Fixed since #612
